### PR TITLE
[mono][jit] Added Vector128 intrinsics that depend on Vector64 for arm64

### PR DESCRIPTION
**Pull request by jandupej**
The methods `Vector128.GetLower,GetUpper,WithLower,WithUpper` and `Create(Vector64,Vector64)` were missing from the intrinsic implementations pending full support of `Vector64`. Now that this has been addressed in https://github.com/dotnet/runtime/pull/86188, we can proceed with this final batch of intrinsics.

Contributes to https://github.com/dotnet/runtime/issues/80566

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -1926,13 +1916,6 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		if (!is_element_type_primitive (fsig->params [0]))
 			return NULL;
 		int op = id == SN_GetLower ? OP_XLOWER : OP_XUPPER;
-
-#ifdef TARGET_AMD64
-		if (!COMPILE_LLVM (cfg))
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:1931]: This is an AMD64 block.

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/mini-arm64.c [hunk 2]
```diff
@@ -4118,6 +4121,25 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 				arm_neon_ext_16b (code, dreg, sreg1, sreg2, ins->inst_c0);
 			break;
 		}
+		case OP_XLOWER: {
+			if (dreg != sreg1)
```
[the remark this answers is not in the document]
Comment (jandupej) [reply src/mono/mono/mini/mini-arm64.c:4125]: The result is 64-bit, so the upper half will not have any effect. However, not sanitizing the upper half may have security implications. Is that a concern?

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (jandupej) [reply src/mono/mono/mini/simd-intrinsics.c:1931]: Good catch. It's easy to confuse `ARM64` and `AMD64` in code.

**Comment by jandupej** [comment]
CI is green, local testing OK. Merging.

**Pull request merged**